Safiyya/circlemapping
# 466 Fix roles being changed across initiatives

### Issue
Fixes #466 

### Description
When a contributor gets added to a circle in the `initiative-helpers-select`, what actually gets added to the circle is the same `Helper` object that exists in the list of options that can be selected. If we then add the same person as a contributor to another circle in the same session, we're again adding that one and the same `Helper` object (i.e. the objects have the same reference, this test passes: `helper1 === helper2`). This means that making changes to that person's role in one circle changes their role in the other circle as well.

The interesting part about this bug is that as soon as we refresh the page, reinitiating the data structures, the two helpers will be deserialised as separate objects, thus fixing the bug. In other words, the bug will only occur when you add the same person to different circles multiple times in the same session but not across sessions.

The fix is to simply create a new `Helper` object whenever a selection is made and add that new object to the circle.

### Setup
In this PR, I've also added notes to the README from when I set the project up locally. There are also new instructions for setting up using a local DB.

### Testing
(for reviewing code, the commits are separated into one with functionality and one with README updates to make it obvious that the actual change to fix #466 is tiny)

1. Check out the `master` branch first to see the bug at work
1. Open/create a new test map (also create some people if you're starting from scratch - this can be done under "Directory")
1. Add two new circles to a test map
1. Add the same contributor to each of the circles
1. Change the role of the contributor in one circle
1. Check that the contributor's role is now also changed in the other circle
1. Add a third circle and add the same person as a contributor
1. Check that roles are now gone for that person across all three circles
1. Now refresh the page and check that the person's roles can now be made different across the circles
1. Now, check out this branch and repeat the above steps, making sure that the roles are now no longer linked (i.e. always behave just like they do after a page refresh / loading an old map)
